acceptance/test_state.go

```diff
@@ -0,0 +1,63 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+)
+
+type testState struct {
+	contextDir string
+}
+
+// Used to set/get state from a context.
+type testStateKey struct{}
```

Comment:
It's odd not to see a constant being used for the key

Reply:
It's a type which I believe it's effectively a thing that doesn't change, like a constant. Also, because it's a type, the only way the context value can be accessed is if someone has access to this particular type. (If another package declares `type testStateKey struct{}` it's considered a different type.)  I like this property.

Godog's [examples](https://github.com/cucumber/godog?tab=readme-ov-file#step-5---add-some-logic-to-the-step-definitions) do seem to guide users towards using types.

Reply:
TIL...